Unreal Engine 5.5 plugin, cannot compile myself

Hi,

we would love to continue using your great plugin - but we cannot compile it for UE 5.5.

There are some errors with the automated test, it claims the ApplicationContextMask is undefined when you are using it, e.g. here:

IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODGrowBuffer_Tests, "ObjectDeliverer.GrowBuffer.Test", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)

That's the error we get:

C:\jetsim-ue55\Plugins\ObjectDeliverer\Source\ObjectDeliverer\Private\Tests\ODGrowBufferTest.cpp(6): error C2838: 'ApplicationContextMask': illegal qualified name in member declaration

But it seems to be defined in AutomationTest.h, via the GetTestFlagsMap.

This is goes way deeper into Unreal Engine C++ than I'm able to understand. 

It would be awesome if you would be able to release a new version.

Best
Andreas

--- a/.vsconfig
+++ b/.vsconfig
@@ -2,9 +2,11 @@
   "version": "1.0",
   "components": [
     "Microsoft.Net.Component.4.6.2.TargetingPack",
-    "Microsoft.VisualStudio.Component.VC.14.34.17.4.x86.x64",
+    "Microsoft.VisualStudio.Component.Unreal.Workspace",
+    "Microsoft.VisualStudio.Component.VC.14.38.17.8.ATL",
+    "Microsoft.VisualStudio.Component.VC.14.38.17.8.x86.x64",
     "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
-    "Microsoft.VisualStudio.Component.Windows10SDK",
+    "Microsoft.VisualStudio.Component.Windows11SDK.22621",
     "Microsoft.VisualStudio.Workload.CoreEditor",
     "Microsoft.VisualStudio.Workload.ManagedDesktop",
     "Microsoft.VisualStudio.Workload.NativeDesktop",

--- a/ObjectDelivererTest.uproject
+++ b/ObjectDelivererTest.uproject
@@ -1,6 +1,6 @@
 {
 	"FileVersion": 3,
-	"EngineAssociation": "5.4",
+	"EngineAssociation": "5.5",
 	"Category": "",
 	"Description": "",
 	"Modules": [

--- a/Plugins/ObjectDeliverer/ObjectDeliverer.uplugin
+++ b/Plugins/ObjectDeliverer/ObjectDeliverer.uplugin
@@ -1,28 +1,27 @@
 {
-	"FileVersion": 3,
-	"Version": 151,
-	"VersionName": "1.5.1",
-	"FriendlyName": "ObjectDeliverer",
-	"Description": "ObjectDeliverer is a data transmission / reception library for Unreal Engine (C ++, Blueprint). ",
-	"Category": "Programming",
-	"CreatedBy": "ayumax",
-	"CreatedByURL": "https://github.com/ayumax",
-	"DocsURL": "https://github.com/ayumax/ObjectDeliverer",
-	"MarketplaceURL": "com.epicgames.launcher://ue/marketplace/content/c2612cff517c48bca7dd080299d97ca8",
-	"SupportURL": "",
-	"CanContainContent": false,
-	"Installed": true,
-	"Modules": [
-		{
-			"Name": "ObjectDeliverer",
-			"Type": "Runtime",
-			"LoadingPhase": "Default",
-			"PlatformAllowList": [
-				"Android",
-				"Win64",
-				"Mac",
-				"IOS"
-			]
-		}
-	]
+  "FileVersion": 3,
+  "Version": 151,
+  "VersionName": "1.5.1",
+  "FriendlyName": "ObjectDeliverer",
+  "Description": "ObjectDeliverer is a data transmission / reception library for Unreal Engine (C ++, Blueprint).",
+  "Category": "Programming",
+  "CreatedBy": "ayumax",
+  "CreatedByURL": "https://github.com/ayumax",
+  "DocsURL": "https://github.com/ayumax/ObjectDeliverer",
+  "MarketplaceURL": "com.epicgames.launcher://ue/marketplace/content/c2612cff517c48bca7dd080299d97ca8",
+  "SupportURL": "https://github.com/ayumax/ObjectDeliverer/issues",
+  "CanContainContent": false,
+  "Modules": [
+    {
+      "Name": "ObjectDeliverer",
+      "Type": "Runtime",
+      "LoadingPhase": "Default",
+      "PlatformAllowList": [
+        "Android",
+        "Win64",
+        "Mac",
+        "IOS"
+      ]
+    }
+  ]
 }

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleSizeBody.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleSizeBody.cpp
@@ -4,7 +4,7 @@
 
 UPacketRuleSizeBody::UPacketRuleSizeBody()
 {
-	
+
 }
 
 UPacketRuleSizeBody::~UPacketRuleSizeBody()
@@ -20,29 +20,15 @@ void UPacketRuleSizeBody::Initialize()
 
 void UPacketRuleSizeBody::MakeSendPacket(const TArray<uint8>& BodyBuffer)
 {
-	auto BodyBufferNum = BodyBuffer.Num();
-	auto SendSize = BodyBufferNum + SizeLength;
-	BufferForSend.SetNum(SendSize, false);
+	const auto BodyBufferNum{ BodyBuffer.Num() };
+	BufferForSend.SetNum(BodyBufferNum + SizeLength, EAllowShrinking::No);
 
-	for (int i = 0; i < SizeLength; ++i)
-	{
-		int32 offset = 0;
-		if (SizeBufferEndian == ECNBufferEndian::Big)
-		{
-			offset = 8 * (SizeLength - i - 1);
-		}
-		else
-		{
-			offset = 8 * i;
-		}
-
-		BufferForSend[i] = (uint8)((BodyBufferNum >> offset) & 0xFF);
-	}
+	for (decltype(SizeLength) i{ 0u }; i < SizeLength; i++)
+		if (BufferForSend.IsValidIndex(i))
+			BufferForSend[i] = (BodyBufferNum >> (SizeBufferEndian == ECNBufferEndian::Big ? 8 * (SizeLength - i - 1) : 8 * i)) & 0xFF;
 
 	FMemory::Memcpy(BufferForSend.GetData() + SizeLength, BodyBuffer.GetData(), BodyBufferNum);
-
 	DispatchMadeSendBuffer(BufferForSend);
-
 }
 
 void UPacketRuleSizeBody::NotifyReceiveData(const TArray<uint8>& DataBuffer)
@@ -52,7 +38,7 @@ void UPacketRuleSizeBody::NotifyReceiveData(const TArray<uint8>& DataBuffer)
 		OnReceivedSize(DataBuffer);
 		return;
 	}
-	
+
 	OnReceivedBody(DataBuffer);
 }
 

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleTerminate.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleTerminate.cpp
@@ -21,53 +21,48 @@ void UPacketRuleTerminate::Initialize()
 
 void UPacketRuleTerminate::MakeSendPacket(const TArray<uint8>& BodyBuffer)
 {
-	BufferForSend.SetNum(BodyBuffer.Num() + Terminate.Num(), false);
-
+	BufferForSend.SetNum(BodyBuffer.Num() + Terminate.Num(), EAllowShrinking::No);
 	FMemory::Memcpy(BufferForSend.GetData(), BodyBuffer.GetData(), BodyBuffer.Num());
 	FMemory::Memcpy(BufferForSend.GetData() + BodyBuffer.Num(), Terminate.GetData(), Terminate.Num());
-
 	DispatchMadeSendBuffer(BufferForSend);
 }
 
 void UPacketRuleTerminate::NotifyReceiveData(const TArray<uint8>& DataBuffer)
 {
 	ReceiveTempBuffer += DataBuffer;
+	const auto TerminateCount{ Terminate.Num() };
 
-	int32 findIndex = -1;
-
-	while (true)
+	while (ReceiveTempBuffer.Num() >= TerminateCount)
 	{
-		for (int i = 0; i <= ReceiveTempBuffer.Num() - Terminate.Num(); ++i)
-		{
-			bool notEqual = false;
-			for (int j = 0; j < Terminate.Num(); ++j)
-			{
-				if (ReceiveTempBuffer[i + j] != Terminate[j])
-				{
-					notEqual = true;
-					break;
-				}
-			}
+		int32 findIndex{ INDEX_NONE };
 
-			if (notEqual == false)
+		for (int32 i{ 0 }, count{ ReceiveTempBuffer.Num() - TerminateCount }; i <= count; i++)
+		{
+			auto bTerminate{ true };
+
+			for (int32 j{ 0 }; j < TerminateCount; j++)
 			{
-				findIndex = i;
+				if (ReceiveTempBuffer[i + j] == Terminate[j])
+					continue;
+
+				bTerminate = false;
 				break;
 			}
+
+			if (!bTerminate)
+				continue;
+
+			findIndex = MoveTemp(i);
+			break;
 		}
 
-		if (findIndex == -1)
-		{
+		if (findIndex <= INDEX_NONE)
 			return;
-		}
 
-		BufferForReceive.SetNum(findIndex, false);
+		BufferForReceive.SetNum(findIndex, EAllowShrinking::No);
 		FMemory::Memcpy(BufferForReceive.GetData(), ReceiveTempBuffer.GetData(), findIndex);
 		DispatchMadeReceiveBuffer(BufferForReceive);
-
 		ReceiveTempBuffer.RemoveAt(0, findIndex + Terminate.Num());
-
-		findIndex = -1;
 	}
 }
 

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleTerminate.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/PacketRule/PacketRuleTerminate.cpp
@@ -21,9 +21,12 @@ void UPacketRuleTerminate::Initialize()
 
 void UPacketRuleTerminate::MakeSendPacket(const TArray<uint8>& BodyBuffer)
 {
-	BufferForSend.SetNum(BodyBuffer.Num() + Terminate.Num(), EAllowShrinking::No);
-	FMemory::Memcpy(BufferForSend.GetData(), BodyBuffer.GetData(), BodyBuffer.Num());
-	FMemory::Memcpy(BufferForSend.GetData() + BodyBuffer.Num(), Terminate.GetData(), Terminate.Num());
+	const auto BodyCount{ BodyBuffer.Num() };
+	const auto TerminateCount{ Terminate.Num() };
+	BufferForSend.SetNum(BodyCount + TerminateCount, EAllowShrinking::No);
+	const auto BufferForSendData{ BufferForSend.GetData() };
+	FMemory::Memcpy(BufferForSendData, BodyBuffer.GetData(), BodyCount);
+	FMemory::Memcpy(BufferForSendData + BodyCount, Terminate.GetData(), TerminateCount);
 	DispatchMadeSendBuffer(BufferForSend);
 }
 
@@ -40,7 +43,7 @@ void UPacketRuleTerminate::NotifyReceiveData(const TArray<uint8>& DataBuffer)
 		{
 			auto bTerminate{ true };
 
-			for (int32 j{ 0 }; j < TerminateCount; j++)
+			for (decltype(i) j{ 0u }; j < TerminateCount; j++)
 			{
 				if (ReceiveTempBuffer[i + j] == Terminate[j])
 					continue;

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolLogReader.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolLogReader.cpp
@@ -74,77 +74,71 @@ void UProtocolLogReader::Close()
 
 void UProtocolLogReader::Send(const TArray<uint8>& DataBuffer) const
 {
-	
+
 }
 
 void UProtocolLogReader::RequestSend(const TArray<uint8>& DataBuffer)
 {
-	
+
 }
 
 bool UProtocolLogReader::ReadData()
 {
-	if (!Reader) return false;
+	if (!Reader)
+		return false;
 
 	while (!Reader->IsEnd() || CurrentLogTime >= 0)
 	{
 		if (CurrentLogTime >= 0)
 		{
-			auto nowTime = (FDateTime::Now() - StartTime).GetTotalMilliseconds();
-			if (CurrentLogTime <= nowTime)
-			{
-				uint32 Size = ReadBuffer.Num();
-				uint32 wantSize = PacketRule->GetWantSize();
+			const auto nowTime{ (FDateTime::Now() - StartTime).GetTotalMilliseconds() };
 
-				if (wantSize > 0)
-				{
-					if (Size < wantSize) return true;
-				}
-
-				int32 Offset = 0;
-				while (Size > 0)
-				{
-					wantSize = PacketRule->GetWantSize();
-					auto receiveSize = wantSize == 0 ? Size : wantSize;
-
-					ReceiveBuffer.SetNum(receiveSize, false);
-
-					int32 Read = 0;
-					FMemory::Memcpy(ReceiveBuffer.GetData(), ReadBuffer.GetData() + Offset, receiveSize);
-					Offset += receiveSize;
-					Size -= receiveSize;
-
-					PacketRule->NotifyReceiveData(ReceiveBuffer);
-				}
-
-				CurrentLogTime = -1;
-			}
-			else
-			{
+			if (CurrentLogTime > nowTime)
 				break;
+
+			auto Size{ ReadBuffer.Num() };
+			auto wantSize{ PacketRule->GetWantSize() };
+
+			if (wantSize > 0 && Size < wantSize)
+				return true;
+
+			auto Offset{ 0u };
+
+			while (Size > 0)
+			{
+				wantSize = PacketRule->GetWantSize();
+				const auto receiveSize{ wantSize == 0 ? Size : wantSize };
+				ReceiveBuffer.SetNum(receiveSize, EAllowShrinking::No);
+				FMemory::Memcpy(ReceiveBuffer.GetData(), ReadBuffer.GetData() + Offset, receiveSize);
+				Offset += receiveSize;
+				Size -= receiveSize;
+				PacketRule->NotifyReceiveData(ReceiveBuffer);
 			}
+
+			CurrentLogTime = -1;
 		}
 
-		if (Reader->RemainSize() < sizeof(double)) return false;
+		if (Reader->RemainSize() < sizeof(double))
+			return false;
 
 		CurrentLogTime = Reader->Read<double>();
+
 		if (IsFirst && CutFirstInterval)
-		{
 			StartTime -= FTimespan::FromMilliseconds(CurrentLogTime);
-		}
+
 		IsFirst = false;
 
-		if (Reader->RemainSize() < sizeof(int32)) return false;
+		if (Reader->RemainSize() < sizeof(int32))
+			return false;
 
-		int32 bufferSize = Reader->Read<int32>();
+		const auto bufferSize{ Reader->Read<int32>() };
 
-		if (Reader->RemainSize() < bufferSize) return false;
+		if (Reader->RemainSize() < bufferSize)
+			return false;
 
-		ReadBuffer.SetNum(bufferSize, false);
-
+		ReadBuffer.SetNum(bufferSize, EAllowShrinking::No);
 		Reader->Read(ReadBuffer, bufferSize);
 	}
-	
 
 	return true;
 }

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolSharedMemory.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolSharedMemory.cpp
@@ -53,7 +53,7 @@ void UProtocolSharedMemory::Start()
 		return;
 	}
 
-	SharedMemoryData = (unsigned char *)MapViewOfFile(SharedMemoryHandle,
+	SharedMemoryData = (unsigned char*)MapViewOfFile(SharedMemoryHandle,
 		FILE_MAP_ALL_ACCESS,
 		0, 0,
 		SharedMemoryTotalSize);
@@ -65,9 +65,9 @@ void UProtocolSharedMemory::Start()
 	}
 
 	ODMutexLock::Lock(SharedMemoryMutex, [this]()
-	{
-		FMemory::Memset(SharedMemoryData, 0, SharedMemoryTotalSize);
-	});
+		{
+			FMemory::Memset(SharedMemoryData, 0, SharedMemoryTotalSize);
+		});
 
 	NowCounter = 0;
 
@@ -86,19 +86,19 @@ void UProtocolSharedMemory::Close()
 {
 #if PLATFORM_WINDOWS
 	ODMutexLock::Lock(SharedMemoryMutex, [this]()
-	{
-		if (!CurrentThread) return;
-		CurrentThread->Kill(true);
+		{
+			if (!CurrentThread) return;
+			CurrentThread->Kill(true);
 
-		delete CurrentThread;
-		CurrentThread = nullptr;
+			delete CurrentThread;
+			CurrentThread = nullptr;
 
-		if (!CurrentInnerThread) return;
-		delete CurrentInnerThread;
-		CurrentInnerThread = nullptr;
+			if (!CurrentInnerThread) return;
+			delete CurrentInnerThread;
+			CurrentInnerThread = nullptr;
 
-		CloseSharedMemory();
-	});
+			CloseSharedMemory();
+		});
 #endif
 
 }
@@ -108,52 +108,54 @@ bool UProtocolSharedMemory::ReceivedData()
 {
 #if PLATFORM_WINDOWS
 
-	uint32 Size = 0;
+	uint32 Size{ 0u };
 
-	ODMutexLock::Lock(SharedMemoryMutex, [this, &Size]()
-	{
-		uint8 counter = 0;
-		FMemory::Memcpy(&counter, SharedMemoryData, sizeof(uint8));
-		if (counter == NowCounter) return;
+	ODMutexLock::Lock
+	(
+		SharedMemoryMutex,
+		[this, &Size]() noexcept
+		{
+			uint8 counter{ 0u };
+			FMemory::Memcpy(&counter, SharedMemoryData, sizeof(uint8));
 
-		NowCounter = counter;
+			if (counter == NowCounter)
+				return;
 
-		FMemory::Memcpy(&Size, SharedMemoryData + sizeof(uint8), sizeof(uint32));
+			NowCounter = counter;
+			FMemory::Memcpy(&Size, SharedMemoryData + sizeof(uint8), sizeof(uint32));
+			TempBuffer.SetNum(Size, EAllowShrinking::No);
+			FMemory::Memcpy(TempBuffer.GetData(), SharedMemoryData + sizeof(uint8) + sizeof(uint32), FMath::Min(SharedMemorySize > 0 ? StaticCast<decltype(Size)>(SharedMemorySize) : 0u, Size));
+		}
+	);
 
-		TempBuffer.SetNum(Size, false);
+	if (Size == 0)
+		return true;
 
-		FMemory::Memcpy(TempBuffer.GetData(), SharedMemoryData + sizeof(uint8) + sizeof(uint32), FMath::Min((uint32)SharedMemorySize, Size));
-	});
+	auto wantSize{ PacketRule->GetWantSize() };
 
-	if (Size == 0) return true;
-	
-	uint32 wantSize = PacketRule->GetWantSize();
+	if (wantSize > 0 && Size < StaticCast<uint32>(wantSize))
+		return true;
 
-	if (wantSize > 0)
-	{
-		if (Size < wantSize) return true;
-	}
+	decltype(Size) Offset{ 0u };
 
-	int32 Offset = 0;
-	while (Size > 0)
+	while (Size > 0u)
 	{
 		wantSize = PacketRule->GetWantSize();
-		auto receiveSize = wantSize == 0 ? Size : wantSize;
-
-		ReceiveBuffer.SetNum(receiveSize, false);
-
-		int32 Read = 0;
+		const auto receiveSize{ wantSize == 0 ? Size : wantSize };
+		ReceiveBuffer.SetNum(receiveSize, EAllowShrinking::No);
 		FMemory::Memcpy(ReceiveBuffer.GetData(), TempBuffer.GetData() + Offset, receiveSize);
 		Offset += receiveSize;
 		Size -= receiveSize;
-
 		PacketRule->NotifyReceiveData(ReceiveBuffer);
 	}
+
 	return true;
+
 #else
+
 	return false;
+
 #endif
-	
 }
 
 void UProtocolSharedMemory::Send(const TArray<uint8>& DataBuffer) const
@@ -176,21 +178,21 @@ void UProtocolSharedMemory::RequestSend(const TArray<uint8>& DataBuffer)
 		int32 writeSize = DataBuffer.Num();
 
 		ODMutexLock::Lock(SharedMemoryMutex, [this, writeSize, &DataBuffer]()
-		{
-			NowCounter++;
-			if (NowCounter == 0)
 			{
-				NowCounter = 1;
-			}
+				NowCounter++;
+				if (NowCounter == 0)
+				{
+					NowCounter = 1;
+				}
 
-			FMemory::Memcpy(SharedMemoryData, &NowCounter, sizeof(uint8));
+				FMemory::Memcpy(SharedMemoryData, &NowCounter, sizeof(uint8));
 
-			FMemory::Memcpy(SharedMemoryData + sizeof(uint8), &writeSize, sizeof(int32));
-			FMemory::Memcpy(SharedMemoryData + sizeof(uint8) + sizeof(int32), DataBuffer.GetData(), writeSize);
+				FMemory::Memcpy(SharedMemoryData + sizeof(uint8), &writeSize, sizeof(int32));
+				FMemory::Memcpy(SharedMemoryData + sizeof(uint8) + sizeof(int32), DataBuffer.GetData(), writeSize);
 
-			FlushViewOfFile(SharedMemoryData, sizeof(uint8) + sizeof(int32) + writeSize);
-		});
-		
+				FlushViewOfFile(SharedMemoryData, sizeof(uint8) + sizeof(int32) + writeSize);
+			});
+
 	}
 #endif
 }

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolSharedMemory.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Protocol/ProtocolSharedMemory.cpp
@@ -116,15 +116,18 @@ bool UProtocolSharedMemory::ReceivedData()
 		[this, &Size]() noexcept
 		{
 			uint8 counter{ 0u };
-			FMemory::Memcpy(&counter, SharedMemoryData, sizeof(uint8));
+			const auto uint8size{ sizeof(uint8) };
+			FMemory::Memcpy(&counter, SharedMemoryData, uint8size);
 
 			if (counter == NowCounter)
 				return;
 
-			NowCounter = counter;
-			FMemory::Memcpy(&Size, SharedMemoryData + sizeof(uint8), sizeof(uint32));
+			NowCounter = MoveTemp(counter);
+			const auto NextSharedMemoryData{ SharedMemoryData + uint8size };
+			const auto uint32size{ sizeof(uint32) };
+			FMemory::Memcpy(&Size, NextSharedMemoryData, uint32size);
 			TempBuffer.SetNum(Size, EAllowShrinking::No);
-			FMemory::Memcpy(TempBuffer.GetData(), SharedMemoryData + sizeof(uint8) + sizeof(uint32), FMath::Min(SharedMemorySize > 0 ? StaticCast<decltype(Size)>(SharedMemorySize) : 0u, Size));
+			FMemory::Memcpy(TempBuffer.GetData(), NextSharedMemoryData + uint32size, FMath::Min(StaticCast<decltype(Size)>(SharedMemorySize), Size));
 		}
 	);
 

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/ODGrowBufferTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/ODGrowBufferTest.cpp
@@ -1,53 +1,53 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "CoreMinimal.h"
+
 #include "Misc/AutomationTest.h"
 #include "../Utils/ODGrowBuffer.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODGrowBuffer_Tests, "ObjectDeliverer.GrowBuffer.Test", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODGrowBuffer_Tests, "ObjectDeliverer.GrowBuffer.Test", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool ODGrowBuffer_Tests::RunTest(const FString& Parameters)
 {
-    const int packetSize = 1024;
+	const int packetSize = 1024;
 
-    {
-        auto buffer = ODGrowBuffer(100);
+	{
+		auto buffer = ODGrowBuffer(100);
 
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(), 100);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize);
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 100);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize);
 
-        buffer.SetLength(2000);
+		buffer.SetLength(2000);
 
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(),  2000);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 2000);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
 
-        uint8 testDataARray[] = { 1, 2, 3 };
-        buffer.Add(ODByteSpan(testDataARray, sizeof(testDataARray)));
+		uint8 testDataARray[] = { 1, 2, 3 };
+		buffer.Add(ODByteSpan(testDataARray, sizeof(testDataARray)));
 
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(), 2003);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
-        TestEqual(TEXT("check Add Data1"), buffer[2000], 1);
-        TestEqual(TEXT("check Add Data2"), buffer[2001], 2);
-        TestEqual(TEXT("check Add Data3"), buffer[2002], 3);
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 2003);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
+		TestEqual(TEXT("check Add Data1"), buffer[2000], 1);
+		TestEqual(TEXT("check Add Data2"), buffer[2001], 2);
+		TestEqual(TEXT("check Add Data3"), buffer[2002], 3);
 
-        buffer.RemoveRangeFromStart(0, 2000);
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(), 3);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
-        TestEqual(TEXT("check Add Data1"), buffer[0], 1);
-        TestEqual(TEXT("check Add Data2"), buffer[1], 2);
-        TestEqual(TEXT("check Add Data3"), buffer[2], 3);
+		buffer.RemoveRangeFromStart(0, 2000);
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 3);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
+		TestEqual(TEXT("check Add Data1"), buffer[0], 1);
+		TestEqual(TEXT("check Add Data2"), buffer[1], 2);
+		TestEqual(TEXT("check Add Data3"), buffer[2], 3);
 
-        uint8 testDataARray2[] = { 0xEE, 0xEF };
-        buffer.CopyFrom(ODByteSpan(testDataARray2, sizeof(testDataARray2)), 1);
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(), 3);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
-        TestEqual(TEXT("check Add Data1"), buffer[0], 1);
-        TestEqual(TEXT("check Add Data2"), buffer[1], 0xEE);
-        TestEqual(TEXT("check Add Data3"), buffer[2], 0xEF);
+		uint8 testDataARray2[] = { 0xEE, 0xEF };
+		buffer.CopyFrom(ODByteSpan(testDataARray2, sizeof(testDataARray2)), 1);
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 3);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
+		TestEqual(TEXT("check Add Data1"), buffer[0], 1);
+		TestEqual(TEXT("check Add Data2"), buffer[1], 0xEE);
+		TestEqual(TEXT("check Add Data3"), buffer[2], 0xEF);
 
-        buffer.Clear();
-        TestEqual(TEXT("check buffer size"), buffer.GetLength(), 0);
-        TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
-    }
+		buffer.Clear();
+		TestEqual(TEXT("check buffer size"), buffer.GetLength(), 0);
+		TestEqual(TEXT("check inner buffer size"), buffer.GetInnerBufferSize(), packetSize * 2);
+	}
 
 	return true;
 }

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/ODJsonSerializerTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/ODJsonSerializerTest.cpp
@@ -1,12 +1,11 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "CoreMinimal.h"
+
 #include "Misc/AutomationTest.h"
+#include "ObjectDelivererManagerTestHelper.h"
 #include "../Utils/JsonSerializer/ODJsonSerializer.h"
 #include "../Utils/JsonSerializer/ODJsonDeserializer.h"
-#include "DeliveryBox/ObjectDeliveryBoxUsingJson.h"
-#include "ObjectDelivererManagerTestHelper.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODJsonSerializer_Tests, "ObjectDeliverer.JsonSerializer.SerializationTest", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODJsonSerializer_Tests, "ObjectDeliverer.JsonSerializer.SerializationTest", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool ODJsonSerializer_Tests::RunTest(const FString& Parameters)
 {
@@ -48,7 +47,7 @@ bool ODJsonSerializer_Tests::RunTest(const FString& Parameters)
 	return true;
 }
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODJsonSerializer_ConvertPropertyNameTest, "ObjectDeliverer.JsonSerializer.ConvertPropertyNameTest", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(ODJsonSerializer_ConvertPropertyNameTest, "ObjectDeliverer.JsonSerializer.ConvertPropertyNameTest", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool ODJsonSerializer_ConvertPropertyNameTest::RunTest(const FString& Parameters)
 {

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleFixedLengthTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleFixedLengthTest.cpp
@@ -1,9 +1,10 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "Misc/AutomationTest.h"
-#include "PacketRule/PacketRuleFixedLength.h"
-#include "PacketRule/PacketRuleFactory.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FCNPacketRuleFixedLengthTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.CNPacketRuleFixedLengthTest.MakeSendPacket", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+#include "Misc/AutomationTest.h"
+#include "PacketRule/PacketRuleFactory.h"
+#include "PacketRule/PacketRuleFixedLength.h"
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FCNPacketRuleFixedLengthTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.CNPacketRuleFixedLengthTest.MakeSendPacket", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FCNPacketRuleFixedLengthTest_MakeSendPacket::RunTest(const FString& Parameters)
 {
@@ -97,7 +98,7 @@ bool FCNPacketRuleFixedLengthTest_MakeSendPacket::RunTest(const FString& Paramet
 	return true;
 }
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FCNPacketRuleFixedLengthTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.CNPacketRuleFixedLengthTest.NotifyReceiveData", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FCNPacketRuleFixedLengthTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.CNPacketRuleFixedLengthTest.NotifyReceiveData", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FCNPacketRuleFixedLengthTest_NotifyReceiveData::RunTest(const FString& Parameters)
 {

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleNodivisionTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleNodivisionTest.cpp
@@ -1,9 +1,10 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "Misc/AutomationTest.h"
-#include "PacketRule/PacketRuleNodivision.h"
-#include "PacketRule/PacketRuleFactory.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleNodivisionTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.PacketRuleNodivisionTest.MakeSendPacket", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+#include "Misc/AutomationTest.h"
+#include "PacketRule/PacketRuleFactory.h"
+#include "PacketRule/PacketRuleNodivision.h"
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleNodivisionTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.PacketRuleNodivisionTest.MakeSendPacket", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleNodivisionTest_MakeSendPacket::RunTest(const FString& Parameters)
 {
@@ -36,7 +37,7 @@ bool FPacketRuleNodivisionTest_MakeSendPacket::RunTest(const FString& Parameters
 	return true;
 }
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleNodivisionTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.PacketRuleNodivisionTest.NotifyReceiveData", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleNodivisionTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.PacketRuleNodivisionTest.NotifyReceiveData", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleNodivisionTest_NotifyReceiveData::RunTest(const FString& Parameters)
 {

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleSizeBodyTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleSizeBodyTest.cpp
@@ -1,9 +1,10 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "Misc/AutomationTest.h"
-#include "PacketRule/PacketRuleSizeBody.h"
-#include "PacketRule/PacketRuleFactory.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleSizeBodyTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.FPacketRuleSizeBodyTest.MakeSendPacket", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+#include "Misc/AutomationTest.h"
+#include "PacketRule/PacketRuleFactory.h"
+#include "PacketRule/PacketRuleSizeBody.h"
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleSizeBodyTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.FPacketRuleSizeBodyTest.MakeSendPacket", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleSizeBodyTest_MakeSendPacket::RunTest(const FString& Parameters)
 {
@@ -74,7 +75,7 @@ bool FPacketRuleSizeBodyTest_MakeSendPacket::RunTest(const FString& Parameters)
 	return true;
 }
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleSizeBodyTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.FPacketRuleSizeBodyTest.NotifyReceiveData", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleSizeBodyTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.FPacketRuleSizeBodyTest.NotifyReceiveData", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleSizeBodyTest_NotifyReceiveData::RunTest(const FString& Parameters)
 {

--- a/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleTerminateTest.cpp
+++ b/Plugins/ObjectDeliverer/Source/ObjectDeliverer/Private/Tests/PacketRuleTerminateTest.cpp
@@ -1,9 +1,10 @@
 // Copyright 2019 ayumax. All Rights Reserved.
-#include "Misc/AutomationTest.h"
-#include "PacketRule/PacketRuleTerminate.h"
-#include "PacketRule/PacketRuleFactory.h"
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleTerminateTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.PacketRuleTerminateTest.MakeSendPacket", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+#include "Misc/AutomationTest.h"
+#include "PacketRule/PacketRuleFactory.h"
+#include "PacketRule/PacketRuleTerminate.h"
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleTerminateTest_MakeSendPacket, "ObjectDeliverer.PacketRuleTests.PacketRuleTerminateTest.MakeSendPacket", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleTerminateTest_MakeSendPacket::RunTest(const FString& Parameters)
 {
@@ -69,7 +70,7 @@ bool FPacketRuleTerminateTest_MakeSendPacket::RunTest(const FString& Parameters)
 	return true;
 }
 
-IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleTerminateTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.PacketRuleTerminateTest.NotifyReceiveData", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FPacketRuleTerminateTest_NotifyReceiveData, "ObjectDeliverer.PacketRuleTests.PacketRuleTerminateTest.NotifyReceiveData", EAutomationTestFlags_ApplicationContextMask | EAutomationTestFlags::SmokeFilter)
 
 bool FPacketRuleTerminateTest_NotifyReceiveData::RunTest(const FString& Parameters)
 {

--- a/Source/ObjectDelivererTest.Target.cs
+++ b/Source/ObjectDelivererTest.Target.cs
@@ -5,11 +5,11 @@ using System.Collections.Generic;
 
 public class ObjectDelivererTestTarget : TargetRules
 {
-	public ObjectDelivererTestTarget(TargetInfo Target) : base(Target)
-	{
-		Type = TargetType.Game;
-		DefaultBuildSettings = BuildSettingsVersion.V2;
-
-		ExtraModuleNames.AddRange( new string[] { "ObjectDelivererTest" } );
-	}
+    public ObjectDelivererTestTarget(TargetInfo Target) : base(Target)
+    {
+        Type = TargetType.Game;
+        DefaultBuildSettings = BuildSettingsVersion.Latest;
+        IncludeOrderVersion = EngineIncludeOrderVersion.Latest;
+        ExtraModuleNames.Add("ObjectDelivererTest");
+    }
 }

--- a/Source/ObjectDelivererTestEditor.Target.cs
+++ b/Source/ObjectDelivererTestEditor.Target.cs
@@ -5,11 +5,11 @@ using System.Collections.Generic;
 
 public class ObjectDelivererTestEditorTarget : TargetRules
 {
-	public ObjectDelivererTestEditorTarget(TargetInfo Target) : base(Target)
-	{
-		Type = TargetType.Editor;
-		DefaultBuildSettings = BuildSettingsVersion.V2;
-
-		ExtraModuleNames.AddRange( new string[] { "ObjectDelivererTest" } );
-	}
+    public ObjectDelivererTestEditorTarget(TargetInfo Target) : base(Target)
+    {
+        Type = TargetType.Editor;
+        DefaultBuildSettings = BuildSettingsVersion.Latest;
+        IncludeOrderVersion = EngineIncludeOrderVersion.Latest;
+        ExtraModuleNames.Add("ObjectDelivererTest");
+    }
 }

